src/services/dom/parser.ts

```diff
@@ -166,7 +180,42 @@ export class Parser {
     if (this.onBeforeParse) {
       this.onBeforeParse(url);
     }
-    const doc = parser.parseFromString(responseText);
+    let doc: Document;
+    try {
+      doc = parser.parseFromString(responseText);
+    } catch (e) {
+      const err = e as Error;
+      const snippet = buildContextSnippet(
+        err.message,
+        responseText,
+        SNIPPET_LENGTH,
+      );
+      if (err instanceof Errors.XMLParserFatalError) {
+        // Re-throw with extra context
+        throw new Errors.ParserFatalError(
+          cleanParserMessage(err.message, snippet),
+          url,
+          snippet,
+          response.status,
+        );
+      }
+      if (err instanceof Errors.XMLParserWarning) {
+        // Re-throw with extra context
+        throw new Errors.ParserWarning(
+          cleanParserMessage(err.message, snippet),
+          url,
+          snippet,
+          response.status,
+        );
+      }
+      // Re-throw with extra context
+      throw new Errors.ParserError(
+        cleanParserMessage(err?.message || 'Unknown error', snippet),
+        url,
+        snippet,
+        response.status,
+      );
```

Comment:
this logic should be wrapped in a try/catch, so that we don't crash trying to process an error